src/js/handlers/createPost.js

```diff
@@ -0,0 +1,61 @@
+// import { createPost } from "../api/posts/create.js";
+import { createPost } from "../api/posts/index.js";
+
+export function setCreatePostFormListener() {
+    const form = document.querySelector("#createPost");
+
+    if (form) {
+        form.addEventListener("submit", async (event) => {
+            event.preventDefault();
+            const formData = new FormData(form);
+            const post = Object.fromEntries(formData.entries());
+
+            const tagsInput = form.querySelector("input[name='tags']").value;
+            const tagsArray = tagsInput.split(",").map((tag) => tag.trim());
+            post.tags = tagsArray;
+
+            console.log("This is a post created", post);
+
+            try {
+                const createdPost = await createPost(post);
+                const createdPostId = createdPost.id;
+                alert("Your post was successfully created.");
+                location.href = `/post/index.html?id=${createdPostId}`;
+            } catch (error) {
+                console.error("Error creating post:", error);
+                alert("An error occurred while creating the post.");
```

Comment:
Maybe the user should see the error message too?

Reply:
Done.